tools/RAiDER/downloadGNSSDelays.py

```diff
@@ -0,0 +1,330 @@
+        if fileurl:
+            results.append({'ID': s, 'year': y, 'path': fileurl})
+    return pd.DataFrame(results).set_index('ID')
+            
+
+def parse_years(input):
+    '''
+    Takes string input and returns a list of years as integers
+    '''
+    test = len(input.split(',')) > 1
+    if test:
+        years = [int(y) for y in input.split(',')]
+        if years[1] > years[0] + 1:
+            years = [years[0] + k for k in range(years[1] - years[0] + 1)]
+    else:
+        years = [int(input)]
+    return years
+    
+
+def parse_args():
+    """Parse command line arguments using argparse."""
+    p = argparse.ArgumentParser(
+          formatter_class=argparse.RawDescriptionHelpFormatter,
+        description="""
+Check for and download tropospheric zenith delays for a set of GNSS stations from UNR
+Usage examples: 
+downloadGNSSdelay.py -y 2010 --check
+downloadGNSSdelay.py -y 2010 -b 40 -79 39 -78 -v
+downloadGNSSdelay.py -y 2010 -f station_list.txt --out products
+""")
+ 
+    p.add_argument(
+        '--years', '-y', dest='years', nargs='+',
+        help="""Year to check or download delays (format YYYY).
+Can be a single value or a comma-separated list. If two years non-consecutive years are given, download each year in between as well. 
+""", type=parse_years, required=True)
+
+    # Stations to check/download
+    area = p.add_argument_group('Stations to check/download. Can be a lat/lon bounding box or file, or will run the whole world if not specified')
+    area.add_argument(
+        '--station_file', '-f', default = None, dest='station_file',
+        help=('Text file containing a list of 4-char station IDs separated by newlines'))
+    area.add_argument(
+        '--BBOX', '-b', dest='bounding_box', type=str, default=None,
+        help="Provide either valid shapefile or Lat/Lon Bounding SNWE. -- Example : '19 20 -99.5 -98.5'")
+
+    misc = p.add_argument_group("Run parameters")
+    misc.add_argument(
+        '--outformat', 
+        help='GDAL-compatible file format if surface delays are requested.',
+        default=None)
+
+    misc.add_argument(
+        '--out', dest='out',
+        help='Directory to download products', 
+        default='.')
+
+    misc.add_argument(
+        '--returntime', dest='returnTime',
+        help="Return delays closest to this specified time. If not specified, the GPS delays for all times will be returned. Input in 'HH:MM:SS', e.g. '16:00:00'", 
+        default=None)
+
+    misc.add_argument(
+        '--download',
+        help='Physically download data. Note this is not necessary to proceed with statistical analysis, as data can be handled virtually in the program.',
+        action='store_true',dest='download', default = False)
+
+    misc.add_argument(
+        '--verbose', '-v',
+        help='Run in verbose (debug) mode? Default False',
+        action='store_true',dest='verbose', default = False)
+
+    return p.parse_args(), p
+
+
+def parseCMD():
```

Comment:
Since this function appears to be the main function for this script, perhaps we should name it something more descriptive? I think this function was originally in the version I wrote so it's my own fault for calling it this!

Reply:
@Askaholic agreed, this is something we would like to do eventually with the other scripts as well. Perhaps next PR though. :)